frontend/__tests__/src/pages/Chapters.test.tsx

```diff
@@ -76,4 +76,30 @@ describe('ChaptersPage Component', () => {
       behavior: 'auto',
     })
   })
+
+  test('renders SearchBar, data, and pagination component concurrently after data is loaded', async () => {
+    window.scrollTo = jest.fn()
+    ;(loadData as jest.Mock).mockResolvedValue({
+      ...mockChapterData,
+      total_pages: 2,
+    })
+    render(<ChaptersPage />)
+
+    const loadingSpinner = screen.getAllByAltText('Loading indicator')
+    await waitFor(() => {
+      expect(loadingSpinner.length).toBeGreaterThan(0)
+      expect(screen.queryByPlaceholderText('Search for OWASP chapters...')).not.toBeInTheDocument()
+      expect(screen.queryByText('Next Page')).not.toBeInTheDocument()
+    })
+    await waitFor(() => {
+      expect(screen.getByPlaceholderText('Search for OWASP chapters...')).toBeInTheDocument()
+      expect(screen.getByText('Chapter 1')).toBeInTheDocument()
+      expect(screen.getByText('Next Page')).toBeInTheDocument()
+
+      const nextPageButton = screen.getByText('Next Page')
+      fireEvent.click(nextPageButton)
```

Comment:
Why do we need this click?

Reply:
@arkid15r if the pagination component is rendered after the data is loaded ...it should be responsive to the click event right ?

Reply:
Checking for presence should be enough. If you fire an event you should check for some expected outcome. Otherwise it's not that obvious (at least to me).

Yes, please remove.

Reply:
sure removed it in the latest commits ...thanks